### ISSUE TITLE
Error when writing: strided data not supported

### ISSUE THREAD

**Issue by paulstillman** [wesm/feather#97]
In python, when I try and write a simple 20x20 dataframe to disk, I keep getting an error stating that strided data isn't yet supported. I don't believe my data is strided (or out of the ordinary), and I can replicate the sample code given on the website without errors, but can't seem to get it to work with my own. Here is some sample code:

```
import feather
import numpy as np
import pandas as pd

tempArr = reshape(np.arange(400), (20,20))
df = pd.DataFrame(tempArr)
feather.write_dataframe(df, 'test.feather')
```

And I get the following error output:

```
Traceback (most recent call last):

  File "<ipython-input-10-85be0b956a3f>", line 1, in <module>
    feather.write_dataframe(testDF, 'test.feather')

  File "/usr/local/lib/python2.7/dist-packages/feather/api.py", line 37, in write_dataframe
    writer.write_array(name, col)

  File "feather/ext.pyx", line 88, in feather.ext.FeatherWriter.write_array (feather/ext.cpp:2127)
    self.write_primitive(name, col, mask)

  File "feather/ext.pyx", line 110, in feather.ext.FeatherWriter.write_primitive (feather/ext.cpp:2432)
    self.write_ndarray(col_values, mask, &values)

  File "feather/ext.pyx", line 135, in feather.ext.FeatherWriter.write_ndarray (feather/ext.cpp:2739)
    check_status(pandas_to_primitive(values, out))

  File "feather/ext.pyx", line 54, in feather.ext.check_status (feather/ext.cpp:1543)
    raise FeatherError(frombytes(c_message))

FeatherError: Invalid: no support for strided data yet

```

I'm on Ubuntu 14.04, running python 2.7.6. Thanks!!

**Comment by wesm** [comment]
Your data is strided because of C-order reshaping (the rows are contiguous / not strided, but not the columns are).

Someone will need to write a patch implementing strided data writes (with tests). I probably will not have time until end of the month.

**Comment by nateGeorge** [comment]
So, if there was a way to transform pandas dataframes to a uniform stride, would that solve the problem too?

**Comment by chris-b1** [comment]
Striding is easier to see with distinct values - e.g.

```
In [65]: arr = np.arange(10); arr
Out[65]: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [66]: arr.reshape((5,2))
Out[66]: 
array([[0, 1],
       [2, 3],
       [4, 5],
       [6, 7],
       [8, 9]])

The column values [0, 2, ...] and [1, 3, ...] are not contiguous in memory.  You can also work around this by taking a copy `df = df.copy()`, which in effect is what your type conversions are doing.

**Comment by nateGeorge** [comment]
Ah, got it, thanks.  Non-strided == contiguous in memory.

**Comment by wesm** [comment]
This is fixed in Feather 0.4 (which depends on pyarrow 0.4.0)
